client/state/selectors/is-eligible-for-upwork-support.ts

```diff
@@ -15,11 +16,7 @@ export default function isEligibleForUpworkSupport( state: AppState ): boolean {
 		return false;
 	}
 
-	const hasBusinessOrEcommercePlan = Object.values( getSitesItems( state ) ).some( ( site ) => {
-		const planSlug = site.plan?.product_slug;
-		return isBusinessPlan( planSlug ?? '' ) || isEcommercePlan( planSlug ?? '' );
-	} );
-
-	// Upwork is not available if the customer has a Business or eCommerce plan
-	return ! hasBusinessOrEcommercePlan;
+	return ! Object.values( getSitesItems( state ) ).some( ( { ID } ) =>
+		siteHasFeature( state, ID ?? 0, WPCOM_FEATURES_UPWORK_SUPPORT_EXEMPT )
+	);
```

Comment:
It took me a moment to understand this logic, may I recommend a comment?

```diff
diff --git a/client/state/selectors/is-eligible-for-upwork-support.ts b/client/state/selectors/is-eligible-for-upwork-support.ts
index 82e3e78a59..e9a79d6741 100644
--- a/client/state/selectors/is-eligible-for-upwork-support.ts
+++ b/client/state/selectors/is-eligible-for-upwork-support.ts
@@ -16,6 +16,8 @@ export default function isEligibleForUpworkSupport( state: AppState ): boolean {
                return false;
        }
 
+       // If any site has the WPCOM_FEATURES_UPWORK_SUPPORT_EXEMPT feature, the
+       // user is not eligible for Upwork Support.
        return ! Object.values( getSitesItems( state ) ).some( ( { ID } ) =>
                siteHasFeature( state, ID ?? 0, WPCOM_FEATURES_UPWORK_SUPPORT_EXEMPT )
        );
```

or

```diff
diff --git a/client/state/selectors/is-eligible-for-upwork-support.ts b/client/state/selectors/is-eligible-for-upwork-support.ts
index 82e3e78a59..f0ab6f3bcb 100644
--- a/client/state/selectors/is-eligible-for-upwork-support.ts
+++ b/client/state/selectors/is-eligible-for-upwork-support.ts
@@ -16,7 +16,8 @@ export default function isEligibleForUpworkSupport( state: AppState ): boolean {
                return false;
        }
 
-       return ! Object.values( getSitesItems( state ) ).some( ( { ID } ) =>
-               siteHasFeature( state, ID ?? 0, WPCOM_FEATURES_UPWORK_SUPPORT_EXEMPT )
+       // To be eligible for Upwork, all of a user's sites must not have the WPCOM_FEATURES_UPWORK_SUPPORT_EXEMPT feature.
+       return Object.values( getSitesItems( state ) ).every(
+               ( { ID } ) => ! siteHasFeature( state, ID ?? 0, WPCOM_FEATURES_UPWORK_SUPPORT_EXEMPT )
        );
 }
```